x-pack/plugins/maps/public/classes/layers/vector_layer/vector_layer.tsx

```diff
@@ -175,6 +181,24 @@ export class VectorLayer extends AbstractLayer implements IVectorLayer {
     });
   }
 
+  async getEditModeEnabled(): Promise<boolean> {
```

Comment:
How about removing getEditModeEnabled from the IVectorLayer interface and moving the logic directly into TOCEntryActionsPopover. The logic is very specific for showing UI parts of TOCEntryActionsPopover and the fewer methods we can put on IVectorLayer the better.